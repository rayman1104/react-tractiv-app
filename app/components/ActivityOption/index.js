/* eslint-disable jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events */
/**
 *
 * ActivityOption
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

const ActivityOption = ({ img, activeImg, text, selected, onSelect }) => (
  <div
    className={`app_main_day_activity ${selected &&
      'app_main_overlay_activity_active'}`}
    onClick={onSelect}
  >
    <div className="app_grid_day_action">
      <img
        src={selected ? activeImg : img}
        className="app_grid_day_action_icon"
        alt={text}
      />
    </div>
    <span className="app_grid_overlay_text">{text}</span>
  </div>
);

ActivityOption.propTypes = {
  text: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  activeImg: PropTypes.string.isRequired,
  selected: PropTypes.bool,
  onSelect: PropTypes.func.isRequired,
};

export default ActivityOption;
