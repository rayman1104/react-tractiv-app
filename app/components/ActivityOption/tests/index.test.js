import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SpinIcon from 'images/icn_spin.svg';
import SpinIconActive from 'images/icn_spin_white.svg';
import ActivityOption from '../index';

describe('<ActivityOption />', () => {
  it('Should render correctly', () => {
    const output = shallow(
      <ActivityOption
        img={SpinIcon}
        activeImg={SpinIconActive}
        text="SPINNING"
        selected
        onSelect={() => {}}
      />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
