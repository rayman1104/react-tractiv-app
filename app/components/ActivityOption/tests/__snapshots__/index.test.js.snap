// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityOption /> Should render correctly 1`] = `
<div
  className="app_main_day_activity app_main_overlay_activity_active"
  onClick={[Function]}
>
  <div
    className="app_grid_day_action"
  >
    <img
      alt="SPINNING"
      className="app_grid_day_action_icon"
      src="IMAGE_MOCK"
    />
  </div>
  <span
    className="app_grid_overlay_text"
  >
    SPINNING
  </span>
</div>
`;
